prediction_market_agent/agents/replicate_to_omen_agent/image_gen.py

```diff
@@ -16,21 +16,26 @@
 
 
 def generate_and_set_image_for_market(
-    market: OmenAgentMarket, api_keys: APIKeys
+    market_address: ChecksumAddress, market_question: str, api_keys: APIKeys
 ) -> IPFSCIDVersion0 | None:
+    market_contract = OmenFixedProductMarketMakerContract(address=market_address)
+    # Test that the market actually exists.
+    # TODO: How can we get the question from the market contract (without using the graph)? (to check that `market_question` is correct)
+    market_contract.totalSupply()
```

Comment:
Delete this code?

Reply:
It will fail if `market_address` doesn't exist, is there a better way to test it than calling a random method?

Reply:
oh i seeeee, makes sense. I don't know if there's a better way to test 😢. But you could do a try-except block, and if it fails, throw a more descriptive exception, like `"Unable to verify contract exists for market address {addr}"`

Reply:
True! Added.